derive_client/_bridge/client.py

```diff
@@ -449,17 +419,78 @@ def _wait_for_lz_settlement(self, pair, from_block, tx_result):
         print(f"🔖 Source [{tx_result.source_chain.name}] OFTSent GUID: {guid.hex()}")
         filter_params = make_filter_params(
             event=pair.target_token.events.OFTReceived(),
-            from_block=from_block,
+            from_block=tx_result.target_from_block,
             argument_filters={"guid": guid},
         )
 
         print(f"🔍 Listening for OFTReceived on [{tx_result.target_chain.name}] at {pair.target_token.address}")
+        return wait_for_event(pair.target_w3, filter_params)
+
+    def fetch_socket_event_log(self, tx_result: BridgeTxResult):
+
+        source_tx = tx_result.source_tx
+        source_w3 = get_w3_connection(tx_result.source_chain)
+        target_w3 = get_w3_connection(tx_result.target_chain)
+
+        abi = json.loads(SOCKET_ABI_PATH.read_text())
+        source_address = SocketAddress[tx_result.source_chain.name].value
+        target_address = SocketAddress[tx_result.target_chain.name].value
+        source_socket = get_contract(source_w3, address=source_address, abi=abi)
+        target_socket = get_contract(target_w3, address=target_address, abi=abi)
+
         try:
-            event_log = wait_for_event(pair.target_w3, filter_params)
-            target_tx.tx_hash = event_log["transactionHash"].to_0x_hex()
-            target_tx.tx_receipt = wait_for_tx_receipt(w3=pair.target_w3, tx_hash=target_tx.tx_hash)
+            source_event = source_socket.events.MessageOutbound().process_log(source_tx.tx_receipt.logs[-2])
+            message_id = source_event["args"]["msgId"]
         except Exception as e:
-            target_tx.exception = e
+            msg = f"Failed to retrieve `msgId` from the Socket MessageOutbound event log from source tx_receipt: {e}"
+            source_tx.exception = ValueError(msg)
+            return tx_result
+
+        print(f"🔖 Source [{tx_result.source_chain.name}] MessageOutbound msgId: {message_id.hex()}")
+        target_event = target_socket.events.ExecutionSuccess()
+        filter_params = target_event._get_event_filter_params(
+            fromBlock=tx_result.target_from_block, abi=target_event.abi
+        )
+
+        def matching_message_id(log: AttributeDict) -> bool:
+            try:
+                decoded = target_event.process_log(log)
+                return decoded["args"].get("msgId") == message_id
+            except Exception:
+                return False
+
+        print(f"🔍 Listening for ExecutionSuccess on [{tx_result.target_chain.name}] at {target_socket.address}")
+        return wait_for_event(target_w3, filter_params, condition=matching_message_id)
+
+    def poll_bridge_progress(self, tx_result: BridgeTxResult) -> BridgeTxResult:
+        # TODO: handle non-pending status
```

Comment:
TODO: here we need to decide what we want. We should probably deal with 
- cases that were already final (SUCCESS, FAILED)
- then for the error case (any non-timeout exception) we might want to allow running this flow. Say there is a `ConnectionError` or some issue with a particular dRPC node: as long as we have the `source_tx.tx_hash` this flow is valid